src/amchart/Pie.js

```diff
@@ -191,10 +191,75 @@
 
         this._chart.validateNow();
         this._chart.validateData();
+        console.log("width: " + this.width());
+        console.log("height: " +this.width());
     };
 
     Pie.prototype.render = function(callback) {
         return HTMLWidget.prototype.render.apply(this, arguments);
+   };
+
+    Pie.prototype.postUpdate = function (domNode, element) {
+         var context = this;
+         
+         var containerBoundingClientRect = context._element.select("svg").node().getBoundingClientRect(); // might need to change this selection (could get a bit more accurate results?)
+         this.d3LabelSelection = element.selectAll(".amcharts-pie-label");
+
+         var left = [];
+         var right = [];
+         var top = [];
+         var bottom = [];
+
+        this.d3LabelSelection.each(function(d, i) {
+            var labelTopPos = d3.select(this).node().getBoundingClientRect().top;
+            if (labelTopPos < containerBoundingClientRect.top) {
+                top.push(labelTopPos - containerBoundingClientRect.top);
+            }
+
+            var labelBottomPos = d3.select(this).node().getBoundingClientRect().bottom;
+            if (labelBottomPos > containerBoundingClientRect.bottom) {
+                bottom.push(containerBoundingClientRect.bottom - labelTopPos);
+
+            }
+
+            var labelRightPos = d3.select(this).node().getBoundingClientRect().right;
+            if (labelRightPos > containerBoundingClientRect.right) {
+                right.push(containerBoundingClientRect.right - labelRightPos);
+            }
+
+
+            var labelLeftPos = d3.select(this).node().getBoundingClientRect().left;
```

Comment:
Only call "getBoundingClientRect" once at the top of the each (its not a "cheap" operation) - also I typically use ```getBBox``` for SVG nodes - I didn't think the bounding rect worked with them (I am not saying its wrong - just thinking out loud).